MapboxCoreNavigation/RouteController.swift

```diff
@@ -653,14 +651,12 @@ extension RouteController: CLLocationManagerDelegate {
         // Force an announcement when the user begins a route
         var alertLevel: AlertLevel = routeProgress.currentLegProgress.alertUserLevel == .none ? .depart : routeProgress.currentLegProgress.alertUserLevel
         var updateStepIndex = false
-        let profileIdentifier = routeProgress.route.routeOptions.profileIdentifier
         
         let userSnapToStepDistanceFromManeuver = distance(along: routeProgress.currentLegProgress.currentStep.coordinates!, from: location.coordinate)
-        let secondsToEndOfStep = userSnapToStepDistanceFromManeuver / location.speed
+        let secondsToEndOfStep = routeProgress.currentLegProgress.currentStepProgress.durationRemaining
```

Comment:
🤔  curious idea. This would require the developer always pass in this annotation option. I guess we could fallback though. Let me check this out. Could be yuge.